src/Microsoft.AspNetCore.Mvc.Core/Routing/ConsumesMatcherPolicy.cs

```diff
@@ -0,0 +1,242 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Formatters;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matchers;
+using Microsoft.AspNetCore.Routing.Patterns;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Mvc.Routing
+{
+    internal class ConsumesMatcherPolicy : MatcherPolicy, IEndpointComparerPolicy, INodeBuilderPolicy
+    {
+        internal const string Http415EndpointDisplayName = "415 HTTP Unsupported Media Type";
+        internal const string AnyContentType = "*/*";
+
+        // Run after HTTP methods, but before 'default'.
+        public override int Order { get; } = -100;
+
+        public IComparer<Endpoint> Comparer { get; } = new ConsumesMetadataEndpointComparer();
```

Comment:
off topic: every time I see this syntax I think it is a getter that is returning a new instance every call

Reply:
![image](https://user-images.githubusercontent.com/1430011/43532142-47372d3c-9566-11e8-8377-2b811ba4aaf9.png)